syncer/sharding-meta/shardmeta.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shardmeta
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pingcap/errors"
+	"github.com/siddontang/go-mysql/mysql"
+
+	"github.com/pingcap/dm/pkg/log"
+	"github.com/pingcap/dm/pkg/utils"
+)
+
+const (
+	// MetaTableFormat is used in meta table name constructor
+	MetaTableFormat = "%s_syncer_sharding_meta"
+)
+
+// DDLItem records ddl information used in sharding sequence organization
+type DDLItem struct {
+	FirstPos mysql.Position `json:"first-pos"` // first DDL's binlog Pos, not the End_log_pos of the event
+	DDLs     []string       `json:"ddls"`      // DDLs
```

Comment:
these ddls have same position? If yes maybe it is better to change the `FirstPos`'s name, and add some comment